src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -518,6 +511,23 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 		sr_dbg("Received %d bytes, %d samples, %d decompressed samples.",
 		       devc->cnt_bytes, devc->cnt_samples,
 		       devc->cnt_samples_rle);
+
+		/*
+		 * The OLS sends its sample buffer backwards.
+		 * Flip it back before sending it on the session bus.
+		 */
+		for (i = 0; i < devc->num_samples / 2; i++) {
+			uint8_t temp[4];
+			memcpy(temp, &devc->raw_sample_buf[4 * i], 4);
+			memmove(&devc->raw_sample_buf[4 * i],
+				&devc->raw_sample_buf[4 * (devc->num_samples -
+							   i - 1)],
+				4);
+			memcpy(&devc->raw_sample_buf[4 * (devc->num_samples -
+							  i - 1)],
```

Comment:
Minor nit: Putting (i - 1) in a seperate line hurts readability IMHO. I'd rather keep it on one line.
The code looks okay. I think the commit message is a bit vague. We mainly want to introduce 'samples_to_write' here, so we can use it for the dynamic buffer later, or did I miss something?

Reply:
OK, I put it on the same line.

I introduce samples_to_write here for clarity, as it's the number of samples that the current sample is expanded to. I think it's easier to understand than "rle_count + 1". The name is debatable, though.